modules/misc/nix.nix

```diff
@@ -10,6 +10,27 @@ let
 
   isNixAtLeast = versionAtLeast (getVersion nixPackage);
 
+  nixPath = lib.concatStringsSep ":" cfg.nixPath;
+
+  useXdg = config.nix.enable
+    && (config.nix.settings.use-xdg-base-directories or false);
+  defexprDir = if useXdg then
+    "${config.xdg.dataHome}/nix/defexpr"
+  else
+    "${config.home.homeDirectory}/.nix-defexpr";
+
+  # The deploy path for declarative channels. The directory name is prefixed
+  # with a number to make it easier for files in defexprDir to control the order
+  # they'll be read relative to each other.
+  channelPath = "${defexprDir}/50-home-manager";
```

Comment:
Is that actually needed? Given that `NIX_PATH` is explicitly set in the module.

Reply:
This is needed to avoid having to specify `-f '<nixpkgs>'` to nix-env every time. nix-env looks for expressions in `~/.nix-defexpr` instead of `NIX_PATH`.

Reply:
This is also how actual Nix channels work.

Reply:
This should be taken care of by `~/.nix-defexpr/channels` linking to `$XDG_STATE_HOME/profiles/channels` according to the manual.

Reply:
`~/.nix-defexpr/channels` is managed by nix-channels. Following the symlink eventually leads to the Nix store, so I can't repurpose it in any way.